Source/SwiftLintFramework/Rules/Lint/EmptyXCTestMethodRule.swift

```diff
@@ -14,34 +14,64 @@ public struct EmptyXCTestMethodRule: Rule, OptInRule, ConfigurationProviderRule
         triggeringExamples: EmptyXCTestMethodRuleExamples.triggeringExamples
     )
 
-    public func validate(file: SwiftLintFile) -> [StyleViolation] {
-        return testClasses(in: file).flatMap { violations(in: file, for: $0) }
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor? {
+        EmptyXCTestMethodRuleVisitor()
+    }
+}
+
+private class EmptyXCTestMethodRuleVisitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+    private(set) var violationPositions: [AbsolutePosition] = []
+
+    override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+        guard let inheritanceList = node.inheritanceClause?.inheritedTypeCollection else {
+            return .skipChildren
+        }
+        let isTestClass = inheritanceList
+            .map(\.typeName)
+            .compactMap { $0.as(SimpleTypeIdentifierSyntax.self) }
+            .map(\.name)
+            .map(\.text)
+            .contains("XCTestCase")
```

Comment:
Two small perf optimizations you could do here:

1. Stop iterating over the inheritance list as soon as an `XCTestCase` type is found
2. Only iterate once

```suggestion
        let isTestClass = inheritanceList.contains { type in
            type.typeName.as(SimpleTypeIdentifierSyntax.self)?.name.text == "XCTestCase"
        }
```

Feel free to break down the closure in more statements, the important thing is to avoid the map/compactMap/map/map/contains to just a contains.

Reply:
This is a habit coming from other languages where the evaluation is always lazy. In Swift we need to start the chain with `.lazy` if the same behavior is desired.

Your suggestion is more concise, however, so I go with it. 😉